docs/blueprints/EP033.rst

```diff
@@ -0,0 +1,252 @@
+
+There are three ways of handling the need for the **proxy port** concept:
+
+1) Not using **proxy ports** at all.
+
+In networks where most or all traffic is transported over inter-switch EVCs (EVCs where UNIs are in different switches), **proxy ports** can be ignored since the `add_int_metadata` action will be executed by the penultimate INT hop only. This is acceptable because all NNI ports would be monitored with INT. However, if intra-EVCs exist in the INT Sink switch, network visibility is compromised without **proxy ports** since intra-EVCs could be responsible for bottlenecks and these bottlenecks wouldn't be detected. This approach can be used when loopback connections are not available on the INT Sink Switch.
+
+2) A shared **proxy port** for many UNIs (1:N).
+
+A share **proxy port** is a solution to address the need for full INT visibility when the amount of total traffic expected by all EVCs isn't higher than the capacity of the **proxy port**. This approach saves physical switch interfaces when users are not heavy users. For instance, consider a INT Sink switch where UNIs don't generate more than 10Gbps over a 100Gbps interface and there are four UNIs. Under normal circumstances, a 100Gbps **proxy port** should suffice for that INT Sink switch since we shouldn't see more than 40Gbps over that **proxy port**. Link aggregation can also be used to increase the **proxy port**'s capacity, for instance, to 200Gbps or more. A combination of #1 (Not using proxy ports) and #2 (Shared proxy port) can be implemented, where all inter-EVCs operate under the premises of #1 and intra-EVCs use the premises of #2.
+
+3) A dedicated **proxy port** per UNI (1:1).
+
+If one UNI along is responsible for 40% or higher of the capacity of a **proxy port**, EP033 suggests a dedicated **proxy port** for the offensive UNI. For instance, a heavy user could have a dedicated **proxy port** while non-heavy users can share a shared **proxy port** or mixing solutions #1, #2, and #3. Dedicated **proxy ports** are recommended for users considered high profile or burst traffic is expected.
+
+
+II. How to use proxy ports
+==========================
+
+As previously described, **proxy ports** are just physical loops placed by the network operators to re-inject packets into the pipeline for further processing. When **proxy ports** are used to enable telemetry, EVCs are broken down in two parts: (1) the source UNI to the **proxy port**, with support for telemetry, and (2) from the **proxy port** to the destination UNI without support for telemetry. The `Telemetry` napp will become responsible for addressing this new scenario as described in EP031.
+
+EP033 provides the following suggestion to integrate the approaches described in Section I with the `telemetry` napp:
+
+  * Leveraging the UNI's Interface object's `metadata` attribute to host a new key: "proxy_port". This new key can have the following values:
```

Comment:
As far as I understand, the Interface object's port_number is the integer that maps to the OpenFlow port number. Isn't it? I don't know interface_id.

Reply:
Yes, correctly. My suggestion is to use Interface.id instead of port_numbers is to be aligned with the way Kytos indexes the interfaces (and also mef_eline refers to the interfaces using the interface_id). Anyway, you can also append the port_number to the switch_id and build the index if required. So I believe that is okay.

Reply:
I will change it to interface.id them. In the end, the telemetry napp will be the one doing all the processing anyway.